projects/packages/my-jetpack/_inc/components/product-detail-card/index.jsx

```diff
@@ -76,12 +76,27 @@ function Price( { value, currency, isOld } ) {
  */
 const ProductDetail = ( { slug, onClick, trackButtonClick } ) => {
 	const { detail, isFetching } = useProduct( slug );
-	const { title, longDescription, features, pricingForUi, isBundle, supportedProducts } = detail;
+	const {
+		title,
+		longDescription,
+		features,
+		pricingForUi,
+		isBundle,
+		supportedProducts,
+		hasRequiredPlan,
+	} = detail;
 
 	const { isFree, fullPrice, currencyCode, discountedPrice } = pricingForUi;
 	const { isUserConnected } = useMyJetpackConnection();
 
-	const addProductUrl = isFree
+	/*
+	 * Product needs purchase when:
+	 * - it's not free
+	 * - it does not have a required plan
+	 */
+	const needsPurchase = ! isFree && ! hasRequiredPlan;
```

Comment:
Wuestion: Which case we have `isFree=true` and `hasRequiredPlan=true`?

Reply:
probably never :-D
But maybe doesn't make sense to define `hasRequiredPlan` when the product is Free either